pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1094,6 +1104,69 @@ type FilesystemMirroringInfoSpec struct {
 	Details string `json:"details,omitempty"`
 }
 
+// FilesystemStatsSpec is the filesystem performance statistics with status update
+type FilesystemStatsSpec struct {
+	// PoolMirroringStatus is the mirroring status of a filesystem
+	// +nullable
+	// +optional
+	*FilesystemStats `json:",inline"`
+	// LastChecked is the last time time the status was checked
+	// +optional
+	LastChecked string `json:"lastChecked,omitempty"`
+	// LastChanged is the last time time the status last changed
+	// +optional
+	LastChanged string `json:"lastChanged,omitempty"`
+	// Details contains potential status errors
+	// +optional
+	Details string `json:"details,omitempty"`
+}
+
+// FilesystemStats represents the filesystem performance statistics
+type FilesystemStats struct {
+	// Version of stats output
+	Version int `json:"version,omitempty"`
+	// GlobalCounters is the list of global performance metrics
+	GlobalCounters []string `json:"global_counters,omitempty"`
+	// Counters is the list of per-mds performance metrics
+	Counters []json.RawMessage `json:"counters,omitempty"`
+	// ClientMetadata are Ceph Filesystem client metadata
+	// +nullable
+	// +optional
+	ClientMetadata map[string]FilesystemClient `json:"client_metadata,omitempty"`
+	// GlobalMetrics are global performance counters
+	// +nullable
+	// +optional
+	GlobalMetrics map[string]GlobalMetricsSpec `json:"global_metrics,omitempty"`
+	// +nullable
+	// +optional
+	// Metrics are per-MDS performance counters (currently, empty) and delayed ranks
+	Metrics *FilesystemMetrics `json:"metrics,omitempty"`
+}
+
+// FilesystemClient represents the client metadata of a mounted Ceph Filesystem
+type FilesystemClient struct {
```

Comment:
🤷🏻  . We get metrics for each client mount yes. See the latest commit message with the example.

Reply:
Let's discuss, exposing the client details in the CR status is too much detail IMO.

Reply:
What's the limit?

Reply:
Etcd has a default limit of [1.5MB](https://github.com/etcd-io/etcd/blob/9c28e07588ffd906bd654b919e58512282c5985b/server/embed/config.go#L57) per object, so if someone creates thousands of volumes it will hit that limit at some point. But even if that limit wasn't there, exposing all the details of individual volumes on the CR just seems too verbose for what should be tracked on a CR status.

Reply:
Agreed. I guess I'll just go ahead and close this then.